### ISSUE TITLE
Windows Support

### ISSUE THREAD

**Issue by Exidex** [project-gauntlet/gauntlet#9]
- [x] Build & Run
- [x] Package
- [x] Applications Plugin. Looks like it requires considerable amount of unsafe Rust code because there are no safe wrappers for the `SHGetKnownFolderItem` & Co APIs

**Comment by Exidex** [comment]
Gave an attempt at implementing Application plugin for Windows, unfortunately couldn't make it work. Application plugin does 2 things: collects list of applications and launches one of those when use selects the item. I was able to successfully collect applications via `SHGetKnownFolderItem` on `FOLDERID_AppsFolder` and iterating over children of that folder where each child is an application. This does work but unfortunately I couldn't find a way to launch an application that would work for different types of those applications.  
- `ShellExecuteExW` seems to work only for url links and exe, but only if PDIL doesn't contain KnownFolder GUID. 
- `IApplicationActivationManager::ActivateApplication` only seems to work for AUMIDs (e.g. `Microsoft.WindowsFeedbackHub_8wekyb3d8bbwe!App`).
- `IContextMenu::InvokeCommand` just fails for unknown reason.

Here is a list of different types of PDILs that I have found in AppsFolder:
- `Microsoft.WindowsFeedbackHub_8wekyb3d8bbwe!App`
- `{F38BF404-1D43-42F2-9305-67DE0B28FC23}\\regedit.exe`
- `{7C5A40EF-A0FB-4BFC-874A-C0F2E0B9FA8E}\\Windows Kits\\10\\Shortcuts\\WindowsStoreAppDevCenterToolsDocumentation.url`
- `{1AC14E77-02E7-4E5D-B744-2EB1AE5198B7}\\WF.msc`
- `{1AC14E77-02E7-4E5D-B744-2EB1AE5198B7}\\appverif.chm`
- `MSEdge`
- `VisualStudio.f7b86bd3`
- `Microsoft.Windows.RemoteDesktop`
- `Microsoft.AutoGenerated.{C804BBA7-FA5F-CBF7-8B55-2096E5F972CB}` - Resource Monitor
- `Blend.f7b86bd3`
- `https://nodejs.org/` - Node.js website

Application needs to be able to launch each of those, either in generic way or by handling each of those separately, but then arises the challenge of being able to reliably tell which of those types the PDIL is.

**Comment by Exidex** [comment]
After investigation, the proper way (which is used by PowerToys Run) seems to be to collect files with following extensions `.lnk`, `.url`, `.exe` and `.appref-ms` in following folders: `FOLDERID_Desktop`, `FOLDERID_PublicDesktop`, `FOLDERID_StartMenu`, `FOLDERID_CommonStartMenu`. Then extract icon using one of `ExtractIconExW`, `ExtractAssociatedIconExW`, `SHGetFileInfoW` (there is pobably more) as close to ~40px as possible and convert to png. Not yet sure about actually executing. 

See for some of that 
- https://github.com/microsoft/PowerToys/blob/main/src/modules/launcher/Plugins/Microsoft.Plugin.Program/Storage/Win32ProgramRepository.cs
- https://github.com/microsoft/PowerToys/blob/main/src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs#L634

But windows api is testing my sanity so I'll leave it for another time

**Comment by Exidex** [comment]
Application plugin was implemented in b406da159840b22e8dbc9df1c1b5191f8a03393c. Leaving this open until release

**Comment by Exidex** [comment]
[v13](https://github.com/project-gauntlet/gauntlet/releases/tag/v13) with application plugin implemented is released, closing this

**Comment by ChillarAnand** [comment]
Gauntlet is opening but nothing is appearing in search in windows.

**Comment by Exidex** [comment]
@ChillarAnand if you are using chocolatey to install Gauntlet, v13 is not yet available there because approval process takes 1 to 3 weeks depending on availability of moderators. Can you please download latest version from GitHub Releases and see if it makes a difference?

**Comment by ChillarAnand** [comment]
I am using a cloud VM that does not allow me to install from the GitHub release msi file.

![Image](https://github.com/user-attachments/assets/d3cb30f9-4287-47df-9ae9-ddd24c372178)

**Comment by Exidex** [comment]
@ChillarAnand I think that is because the .msi file is not signed. Not sure if it ever will be, that certificate is expensive.

As a workaround you could try unblocking this specific via properties ui 

> 1. Right click on installer.msi
> 2. Select properties option.
> 3. Click on checkbox to check Unblock at the bottom of Properties

Or if you are familiar with powershell you could try using [Unblock-File](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/unblock-file)

Try it out, and let me know if it works

**Comment by ChillarAnand** [comment]
Thanks for the tip, @Exidex. I have uninstalled existing choco install and am able to install via msi file now. 

However, on opening the app, I am not able to search any apps or not able to use it as calculator. I couldn't any docs related to any additional setup as well.

<img width="744" alt="Image" src="https://github.com/user-attachments/assets/403e8136-f727-4f7d-b818-fba10bdb58fc" />

<img width="744" alt="Image" src="https://github.com/user-attachments/assets/f29f3010-fe4e-435c-9547-19cf21165128" />

**Comment by Exidex** [comment]
@ChillarAnand Please create a new issue and specify os and gauntlet version
